docs/resources/integration_aws_external_id.md

```diff
@@ -0,0 +1,41 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "datadog_integration_aws_external_id Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  !>A new external ID must be used to create an AWS account integration within 48 hours of creation or it will expire.
+  !>Running terraform destroy only removes the resource from Terraform state and does not deactivate anything in Datadog or AWS.
+  Provides a Datadog-Amazon Web Services external ID resource. This can be used to create Datadog-Amazon Web Services external IDs
+---
+
+# datadog_integration_aws_external_id (Resource)
+
+!>A new external ID must be used to create an AWS account integration within 48 hours of creation or it will expire.
+
+!>Running `terraform destroy` only removes the resource from Terraform state and does not deactivate anything in Datadog or AWS.
+
+Provides a Datadog-Amazon Web Services external ID resource. This can be used to create Datadog-Amazon Web Services external IDs
```

Comment:
nit/suggestion: Could we add some instructions / an example of how this would be used in conjunction with the `datadog_integration_aws_account` resource? Maybe just link to the resource and note that you can reference this external ID value as the value of `auth_config.aws_auth_config_role.external_id` like 
```
resource "datadog_integration_aws_account" "foo-defaults" {
  aws_account_id = "123456789019"
  aws_partition  = "aws"

  auth_config {
    aws_auth_config_role {
      role_name   = "DatadogIntegrationRole"
      external_id = datadog_integration_aws_external_id.foo.id
    }
  }
```
And then maybe something about forcing a new value to regenerate? Like 
```
terraform apply -replace datadog_integration_aws_external_id.foo
```